core/client/fs/src/main/java/alluxio/client/RetryHandlingMetaMasterClient.java

```diff
@@ -68,6 +72,18 @@ protected void afterConnect() {
     mClient = new MetaMasterClientService.Client(mProtocol);
   }
 
+  @Override
+  public synchronized List<ConfigProperty> getConfiguration() throws IOException {
+    return retryRPC(() -> {
+      List<ConfigProperty> configList = new ArrayList<>();
+      for (alluxio.thrift.ConfigProperty configInfo : mClient
```

Comment:
NIT: in java 8, you can achieve list comprehension by

```java
configList =  mClient.getConfiguration(new GetConfigurationTOptions()).getConfigList()
    .stream()
    .map(configInfo -> ConfigProperty.fromThrift(configInfo))
    .collect(toList()));
```

Reply:
Great suggestion, I will be more and more getting used to lambda and stream and using them more in the future! Improved, thanks!